lib/ash/code_interface.ex

```diff
@@ -449,60 +503,25 @@ defmodule Ash.CodeInterface do
 
         interface_options = Ash.Resource.Interface.interface_options(action.type, interface)
 
-        resolve_opts_params =
+        resolve_params_and_opts =
           quote do
             {params, opts} =
-              if opts == [] && Keyword.keyword?(params_or_opts),
-                do:
-                  {%{},
-                   unquote(interface_options).validate!(params_or_opts)
-                   |> unquote(interface_options).to_options()},
-                else:
-                  {params_or_opts,
-                   unquote(interface_options).validate!(opts)
-                   |> unquote(interface_options).to_options()}
+              Ash.CodeInterface.params_and_opts(params_or_opts, opts, fn opts ->
+                unquote(Macro.escape(interface.default_options))
+                |> Keyword.merge(opts)
```

Comment:
```suggestion
                |> Keyword.merge(opts, fn :bulk_options, v1, v2 -> 
                  Keyword.merge(List.wrap(v1), List.wrap(v2))
                  
                  _key, v1, v2 ->
                    v2
                end)
```

This is the change I'm suggesting.

This way, if the interface sets `default_options: [bulk_options: [strategy: :stream]]` and the user specifies `bulk_options: [batch_size: 500]`, the value of `bulk_options` will be `[strategy: :stream, batch_size: 500]`, not `[batch_size: 500]`

Reply:
Hmmm. Is there anything else that might cause a similar issue where it'd make sense to deep merge?

Reply:
Probably also the `page` parameter and `load` parameter. These are more complex though, as `page` can be `false`. So if the users value is `false` we should overwrite the default, otherwise we should merge.

for `load` we should just be able to do something like `List.wrap(default) ++ List.wrap(new)`

Reply:
Time to write a function, lol.

Reply:
Written. This is turning into a lot more than I anticipated. I added a bunch of doctests too which... caught a bug.